product-service/src/libs/data-service.ts

```diff
@@ -1,10 +1,118 @@
-import { productsData } from '@mocks/products';
-import { IProduct, IProductsData } from 'src/types/api-types';
+import { Product, ProductsData, Stock } from 'src/types/api-types';
+import {
+  AttributeValue,
+  DynamoDBClient,
+  PutItemCommand,
+  PutItemCommandInput,
+  QueryCommand,
+  QueryInput,
+  ScanCommand,
+  ScanCommandInput
+} from '@aws-sdk/client-dynamodb';
 
-export const getProducts = async (): Promise<IProductsData> => {
-  return Promise.resolve(productsData);
+const dynamodb = new DynamoDBClient({ region: process.env.REGION || 'us-east-1' });
+
+function deserializeDynamoDBItem(item: Record<string, AttributeValue>) {
+  const deserializedItem = {};
+  for (const [attributeName, value] of Object.entries(item)) {
+    const attributeType = Object.keys(value)[0];
+
+    if (attributeType === 'S') {
+      deserializedItem[attributeName] = value.S;
+    } else if (attributeType === 'N') {
+      deserializedItem[attributeName] = parseFloat(value.N);
+    }
+  }
+
+  return deserializedItem;
+}
+
+async function queryProducts(): Promise<Product[]> {
+  const params: ScanCommandInput = {
+    TableName: process.env.PRODUCTS_TABLE_NAME || 'products'
+  };
+
+  const data = await dynamodb.send(new ScanCommand(params));
+
+  return data.Items.map((item) => deserializeDynamoDBItem(item)) as Product[];
```

Comment:
**Question**: Total count of items does not interesting you?

Reply:
I don't know if it should at the moment :) I didn't consider we'll have pagination so I didn't do any prep for that (though had in mind that the DB might have a lot of items etc, and this topic was brought at the last Q&A session)

Reply:
added actually, can be useful, thanks